### ISSUE TITLE
getAsString with an optional string argument in JsonObject

### ISSUE THREAD

**Issue by faissaloo** [google/gson#1291]
So I don't have to use ```.get("tag").getAsString()``` and have to deal with ```.get("tag")``` being null.

**Comment by lyubomyr-shaydariv** [comment]
I would cause API bloating: this would require the entire `get***(name)` and `is***(name)` method family to be replicated in `JsonObject` and `get***(index)` and `is***(index)`  to be replicated in `JsonArray` respectively. Having `.get(name)`/`.get(index)` is simpler to design and guarantees that you can always access whole `JsonElement` API not worrying that some APIs are not replicated to `JsonObject` or `JsonArray`.

**Comment by philipwhiuk** [comment]
I agree it causes API bloating however I disagree that it’s a problem here. The point of the API is to provide easy access to JSON.

Not implementing this just causes implementors to reimplement the null check themselves - which is obvious sub par compared to a tested library implementation.

It would be nice if Java had an ability y where you could mix in the extra methods if you wanted the convenience. But we don’t have that in the language so we are forced to compromise.

My personal use case is providing this as an API for downstream developers to access custom defined structures so this is a common case which I’d like to improve to provide an API that is more accessible. If forking is the only way to implement it that would be annoying.
